Main.hs

```diff
@@ -200,45 +183,48 @@ handleEvent (MessageUpdate Message{..}) = do
                              , oldreq /= req
                              -> if null req
                                 then case reply of
-                                  Just id -> do sendTrace $ DeleteMessage (messageChannel, id)
+                                  Just id -> do void $ restCallTrace $ DeleteMessage (messageChannel, id)
```

Comment:
Is the `void` here really necessary? I liked my auxiliary function more.

Reply:
`when :: Bool -> f () -> f (). I guess it is a style difference. `send` and `sendTrace` seem so similar that I prefer to combine them into the more generic version and then massage any `f a` into `f ()` when i need to.